### PR TITLE
Ms graph user permissions bug fix

### PR DESCRIPTION
**Pull request by yucohen**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/47067)

## Description
Added the client credentials flow as authentication method for the self deployed configuration, since the authorization code flow did not work properly for the msgraph-user-session-revoke command (when authenticated with a user with no admin roles)
## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@itay4 Doc review completed.

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md [hunk 1]
```diff
@@ -13,17 +9,30 @@ You need to grant Cortex XSOAR authorization to access Azure Active Directory Us
 
 ## Authorize Cortex XSOAR for Azure Active Directory Users (self-deployed configuration)
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md:10]:
```suggestion
## Authorize Cortex XSOAR for Azure Active Directory Users (Self deployed Azure App)
```

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md [hunk 2]
```diff
@@ -1,7 +1,3 @@
-For full integration documentation, see our [reference docs](https://xsoar.pan.dev/docs/reference/integrations/microsoft-graph-user).
-
-The commands: ***msgraph-user-terminate-session***, ***msgraph-user-change-password*** are only supported in a self-deployed application configuration.
-
 
 ## Authorize Cortex XSOAR for Azure Active Directory Users
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md:2]:
```suggestion
## Authorize Cortex XSOAR for Azure Active Directory Users (Cortex XSOAR Azure App)
```

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md [hunk 3]
```diff
@@ -13,17 +9,30 @@ You need to grant Cortex XSOAR authorization to access Azure Active Directory Us
...
+
+In order to use the ***msgraph-user-change-password*** command, you must configure with the authorization code flow.
+
+Note: When using the authorization code flow, make sure the user you authenticate with has the correct roles in Azure AD in order to use the command.
 
-1. To use a self-configured Azure application, you need to add a new Azure App Registration in the Azure Portal. To add the registration, refer to the following [Microsoft documentation](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).
-2. Make sure the following permissions are granted for the app registration:
-   -  API/Permission name `Directory.AccessAsUser.All` of type `Delegated`
-3. Copy the following URL and replace the ***TENANT_ID***, ***CLIENT_ID*** and ***REDIRECT_URI*** with your own client ID and redirect URI, accordingly.
+### Self deployed configuration with client credentials flow
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md:22]: this and the below section are already documented in our ms auth in xsoar.pan.dev, lets just refer to there

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md [hunk 4]
```diff
@@ -13,17 +9,30 @@ You need to grant Cortex XSOAR authorization to access Azure Active Directory Us
 
 ## Authorize Cortex XSOAR for Azure Active Directory Users (self-deployed configuration)
 
-Follow these steps for a self-deployed configuration.
+There are two different authentication methods for self-deployed configuration: 
+- [client credentials flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow)
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md:13]:
```suggestion
- [Client Credentials flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow)
```

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md [hunk 5]
```diff
@@ -13,17 +9,30 @@ You need to grant Cortex XSOAR authorization to access Azure Active Directory Us
...
+- [authorization code flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow).
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md:14]:
```suggestion
- [Authorization Code flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow)
```

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md [hunk 6]
```diff
@@ -13,17 +9,30 @@ You need to grant Cortex XSOAR authorization to access Azure Active Directory Us
...
+
+We recommend using the client credentials flow.
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md:16]:
```suggestion
We recommend using the Client Credentials flow.
```

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md [hunk 7]
```diff
@@ -13,17 +9,30 @@ You need to grant Cortex XSOAR authorization to access Azure Active Directory Us
...
+
+In order to use the ***msgraph-user-change-password*** command, you must configure with the authorization code flow.
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md:18]:
```suggestion
In order to use the ***msgraph-user-change-password*** command, you must use with the Authorization Code flow.
```

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md [hunk 8]
```diff
@@ -13,17 +9,30 @@ You need to grant Cortex XSOAR authorization to access Azure Active Directory Us
...
+
+Note: When using the authorization code flow, make sure the user you authenticate with has the correct roles in Azure AD in order to use the command.
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser_description.md:20]:
```suggestion
**Note:** When using the Authorization Code flow, make sure the user you authenticate with has the relevant roles in Azure AD in order to execute the operation.
```

File: Packs/MicrosoftGraphUser/ReleaseNotes/1_4_2.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Azure Active Directory Users
+- Added *client credentials flow* as another authentication flow for a self-deployed configuration.
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/ReleaseNotes/1_4_2.md:4]:
```suggestion
- Added the Client Credentials authentication flow for self deployed Azure App configuration.
```

File: Packs/MicrosoftGraphUser/ReleaseNotes/1_4_2.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+- Fixed an issue where the ***msgraph-user-session-revoke*** command did not work properly with the *authorization code flow*.
```
Comment (Itay4) [inline Packs/MicrosoftGraphUser/ReleaseNotes/1_4_2.md:5]: i think we can remove the second bullet

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py [hunk 11]
```diff
@@ -218,8 +218,6 @@ def test_function(client, _):
             # for self deployed app
             raise Exception("When using a self-deployed configuration, "
                             "Please enable the integration and run the !msgraph-user-test command in order to test it")
-        if not demisto.params().get('auth_code') or not demisto.params().get('redirect_uri'):
```
[the remark this answers is not in the document]
Comment (yucohen) [reply Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py:224]: correct
deleted this line and added a check to the main

**Review by Itay4 — APPROVED** [review]

**Pull request merged**